app/forms/idv/finance_form.rb

```diff
@@ -44,11 +44,13 @@ def initialize(idv_params)
     def submit(params)
       @params = params
       finance_value = update_finance_values(params)
-      return false unless valid?
 
-      clear_idv_params_finance
-      idv_params[finance_type] = finance_value
-      true
+      if valid?
```

Comment:
What do you think about assigning `valid?` to a variable, like `success`, and then using the variable, to avoid calling `valid?` twice?